src/Buf.mo

```diff
@@ -131,6 +143,8 @@ public class Buf<X> (initCapacity : Nat) {
     }
   };
 
+  /// Overwrites the 0-indexed Element for `offset` with `elem`. Traps if the
+  /// index is out of bounds.
   public func put(offset : Nat, elem : X) {
     elems[offset] := elem;
   };
```

Comment:
```suggestion
  /// Overwrites the current value of the `i`-entry of  this buffer with `elem`. Traps if the
  /// index is out of bounds. Indexing is zero-based.
  public func put(i : Nat, elem : X) {
    elems[i] := elem;
  };
```
Shouldn't this either trap if i >= count or grow the Buffer? Otherwise its behaviour depends on the current capacity.

Reply:
Good question --- But I think any choice raises questions: Suppose we choose to grow (not trap), then there isn't an "obvious" way to fill in the gap that may be defined.  How do you choose the otherwise-missing elements between the current last one, and this one?  Does the user want to replicate the `put` element?  The current last element (if any)?  (What did you have in mind?)